### DISCUSSION TITLE
How to let other component capture click while Dialog is open?

### DISCUSSION THREAD

**Discussion by QzCurious** [radix-ui/primitives#1877] [Help] [answered]
This question is similar to #1720.

## Current Behavior

My app might pop a toast at any time. And the toast could be closed by click on it.
However, if a Dialog is opening, the Dialog is preventing my toast to be closed.

Here is the simulation:

https://user-images.githubusercontent.com/38932402/211143365-9e9fe8e2-cf92-4a35-9158-191bd6dfcb48.mp4

## What I had tried

1. `modal={false}` so that my toast can capture to click. But when I click toast, the Dialog is closed too.
2. As [answer](https://github.com/radix-ui/primitives/discussions/1720#discussioncomment-3877058) on #1720. By applying `pointer-event: auto` to my toast. I can capture the click. But the Dialog is closed too.

## What I Want

How can I make the click once per element. That is, first click close toast and second click close Dialog.

## CodeSandbox

This [example code](https://codesandbox.io/s/dialog-prevent-everything-from-being-click-0sk86f?file=/App.js) is a tweak version from the [documentation](https://www.radix-ui.com/docs/primitives/components/dialog).

The tweaks:
I've only add a `div` with `onClick` to simulate the behavior.

**Comment by benoitgrelard** [comment, marked as answer]
With 2. you are mostly there, all you'd need to do is make sure you don't close when interacting outside the dialog if the interaction is inside the toast region. You can use a data attribute or classname or anything you want to check the origin of the event, and call `event.preventDefault()` in `onPointerDownOutside` when that matches.

**Reply by QzCurious** [reply]
`onPointerDownOutside` hits me that the closing behavior is determined by click outside of [`<Dialog.Content />`](https://www.radix-ui.com/docs/primitives/components/dialog#content).
